docs/app-host/executable-resources.md

```diff
@@ -0,0 +1,277 @@
+---
+title: Host external executables in .NET Aspire
+description: Learn how to use ExecutableResource and AddExecutable to host external executable applications in your .NET Aspire app host.
+ms.date: 01/04/2025
+---
+
+# Host external executables in .NET Aspire
+
+In .NET Aspire, you can host external executable applications alongside your .NET projects using the <xref:Aspire.Hosting.ExecutableResourceBuilderExtensions.AddExecutable%2A> method. This capability is useful when you need to integrate non-.NET applications or tools into your distributed application, such as Node.js applications, Python scripts, or specialized CLI tools.
+
+## When to use executable resources
+
+Use executable resources when you need to:
+
+- Host non-.NET applications that don't have containerized equivalents
+- Integrate command-line tools or utilities into your application
+- Run external processes that other resources depend on
+- Develop with tools that provide local development servers
+
+Common examples include:
+
+- **Frontend development servers**: Tools like Azure Static Web Apps CLI, Vite, or webpack dev server
+- **Language-specific applications**: Node.js apps, Python scripts, or Go applications
+- **Database tools**: Migration utilities or database seeders
+- **Build tools**: Asset processors or code generators
+
+## Basic usage
+
+The <xref:Aspire.Hosting.ExecutableResourceBuilderExtensions.AddExecutable%2A> method requires a resource name, the executable path, and optionally command-line arguments and a working directory:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Basic executable without arguments
+var nodeApp = builder.AddExecutable("frontend", "node", "server.js", ".");
+
+// Executable with command-line arguments
+var pythonApp = builder.AddExecutable("api", "python", "-m", "uvicorn")
+    .WithArgs("main:app", "--reload", "--host", "0.0.0.0", "--port", "8000");
+
+builder.Build().Run();
+```
+
+### Method parameters
+
+The <xref:Aspire.Hosting.ExecutableResourceBuilderExtensions.AddExecutable%2A> method accepts the following parameters:
+
+- **name**: A unique identifier for the resource
+- **command**: The executable name or path (for example, `"node"`, `"python"`, or `"/usr/bin/myapp"`)
+- **workingDirectory**: The directory where the executable should run
+- **args**: Optional command-line arguments passed to the executable
+
+## Configure command-line arguments
+
+You can provide command-line arguments in several ways:
+
+### Arguments in the AddExecutable call
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Arguments provided directly
+var app = builder.AddExecutable("swa-cli", "swa", "start", ".")
```

Comment:
Cool, but instead of doing Azure Static Web Apps CLI, please base this on either [Vercel](https://vercel.com/docs/cli) or [Netlify](https://docs.netlify.com/api-and-cli-guides/cli-guides/get-started-with-cli/). With a link to their docs.